app/src/main/java/com/google/samples/apps/sunflower/compose/PlantDetailView.kt

```diff
@@ -0,0 +1,505 @@
+            description = plant.description,
+            onNamePositioned = {
+                if (namePosition == Float.MAX_VALUE) {
+                    namePosition = it.globalPosition.y
+                }
+            },
+            toolbarShown = toolbarShown
+        )
+    }
+}
+
+@Composable
+private fun PlantHeader(
+    toolbarShown: Boolean,
+    plantName: String,
+    callbacks: PlantDetailsCallbacks,
+    transitionState: TransitionState
+) {
+    if (toolbarShown) {
+        PlantDetailsToolbar(
+            plantName = plantName,
+            onBackClicked = callbacks.onBackClicked,
+            onShareClicked = callbacks.onShareClicked,
+            modifier = Modifier.drawOpacity(transitionState[toolbarAlphaKey])
+        )
+    } else {
+        PlantHeaderActions(
+            onBackClicked = callbacks.onBackClicked,
+            onShareClicked = callbacks.onShareClicked,
+            modifier = Modifier.drawOpacity(transitionState[contentAlphaKey])
+        )
+    }
+}
+
+@Composable
+private fun PlantDetailsToolbar(
+    plantName: String,
+    onBackClicked: () -> Unit,
+    onShareClicked: () -> Unit,
+    modifier: Modifier = Modifier
+) {
+    Column(modifier) {
+        val spacerHeight = with(DensityAmbient.current) {
+            InsetsAmbient.current.systemBars.top.toDp()
+        }
+        Spacer(
+            Modifier.preferredHeight(spacerHeight).fillMaxWidth()
+                .drawBackground(MaterialTheme.colors.surface)
+        )
+        TopAppBar(
+            backgroundColor = MaterialTheme.colors.surface
+        ) {
+            IconButton(onBackClicked, Modifier.gravity(Alignment.CenterVertically)) {
+                Icon(Icons.Filled.ArrowBack)
+            }
+            Text(
+                text = plantName,
+                color = MaterialTheme.colors.onSurface,
+                style = MaterialTheme.typography.h6,
+                modifier = Modifier.weight(1f).fillMaxSize()
+                    .wrapContentSize(Alignment.Center)
+            )
+            val shareAccessibilityLabel = stringResource(R.string.menu_item_share_plant)
+            IconButton(
+                onShareClicked,
+                Modifier.gravity(Alignment.CenterVertically).semantics {
+                    accessibilityLabel = shareAccessibilityLabel
+                }
+            ) {
+                Icon(Icons.Filled.Share)
+            }
+        }
+    }
+}
+
+@Composable
+private fun PlantImageHeader(
+    scrollerPosition: ScrollerPosition,
+    imageUrl: String,
+    onFabClicked: () -> Unit,
+    isPlanted: Boolean,
+    modifier: Modifier = Modifier,
+    transitionModifier: Modifier = Modifier
+) {
+    val imageHeight = state { 0 }
+
+    Stack(modifier.fillMaxWidth()) {
+        PlantImage(scrollerPosition, imageUrl, transitionModifier.onPositioned {
+            imageHeight.value = it.size.height
+        })
+        if (!isPlanted) {
+            val fabModifier = if (imageHeight.value != 0) {
```

Comment:
So the FAB disappears instantaneously when entering 'collapsed' mode? Is that a regression from Views? Doesn't it animate it's scale to hide/show? Could use a transition here?

Reply:
It uses the `transitionModifier` so it will be fading out with image